core/src/main/scala/org/http4s/Message.scala

```diff
@@ -292,13 +391,32 @@ sealed abstract case class Request[F[_]](
   /** Whether the Request was received over a secure medium */
   lazy val isSecure: Option[Boolean] = connectionInfo.map(_.secure)
 
-  def serverSoftware: ServerSoftware =
+  lazy val serverSoftware: ServerSoftware =
     attributes.get(Keys.ServerSoftware).getOrElse(ServerSoftware.Unknown)
 
   def decodeWith[A](decoder: EntityDecoder[F, A], strict: Boolean)(f: A => F[Response[F]])(
       implicit F: Monad[F]): F[Response[F]] =
     decoder.decode(this, strict = strict).fold(_.toHttpResponse[F](httpVersion), f).flatten
 
+  /** Helper method for decoding [[Request]]s
+    *
+    * Attempt to decode the [[Request]] and, if successful, execute the continuation to get a [[Response]].
+    * If decoding fails, an `UnprocessableEntity` [[Response]] is generated.
+    */
+  final def decode[A](
```

Comment:
I'm skeptical of the value of this and `decodeStrict`, but more people use them than I realize.  I don't know that I want to fight the deprecation battle for something so common, but I dislike having multiple ways to do it.

Reply:
I would be 👍 on deprecating it. I guess we can look at reports for it for the milestones as a gauge for how much it's used.

Reply:
Deprecation + scalafix is going to be a powerful combination for us for things that can be translated directly.  It would be nice if the two start going together once the scalafix lands.  I'm comfortable taking this on in a subsequent PR.